java/client/src/main/java/glide/internal/GlideCoreClient.java

```diff
@@ -0,0 +1,1100 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.internal;
+
+import glide.api.OpenTelemetry;
+import glide.api.models.exceptions.ClosingException;
+import glide.ffi.resolvers.NativeUtils;
+import glide.ffi.resolvers.OpenTelemetryResolver;
+import glide.internal.protocol.*;
+import java.lang.ref.Cleaner;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * GLIDE core client transport using JNI. Replaces UDS-based communication. Provides direct JNI
+ * access to glide-core with all routing and performance optimizations.
+ */
+public class GlideCoreClient implements AutoCloseable {
+    private static final Cleaner CLEANER = Cleaner.create();
+
+    static {
+        // Load the native library
+        try {
+            NativeUtils.loadGlideLib();
+        } catch (Exception e) {
+            // Use proper logging instead of System.err.println
+            glide.api.logging.Logger.log(
+                    glide.api.logging.Logger.Level.ERROR,
+                    "GlideCoreClient",
+                    "Failed to load native library: " + e.getMessage());
+            throw new RuntimeException("Failed to load native library", e);
+        }
+        onNativeInit();
+    }
+
+    private static native void onNativeInit();
+
+    private static native void freeNativeBuffer(long id);
+
+    private static final java.util.concurrent.ConcurrentHashMap<
+                    Long, java.lang.ref.WeakReference<glide.api.BaseClient>>
+            clients = new java.util.concurrent.ConcurrentHashMap<>();
+
+    private static final byte[][] EMPTY_SCRIPT_PARAMS = new byte[0][];
```

Comment:
Rename to be not script-specific and use elsewhere when we need an empty 2D byte array (eg subPattern).